test/ctia/entity/target_record_test.clj

```diff
@@ -0,0 +1,96 @@
+                                                  new-target-record-maximal]]))
+
+(use-fixtures :once (join-fixtures [mth/fixture-schema-validation
+                                    helpers/fixture-properties:clean
+                                    whoami-helpers/fixture-server]))
+
+(use-fixtures :each whoami-helpers/fixture-reset-state)
+
+(defn additional-tests [_ target-record-sample]
+  (testing "GET /ctia/target-record/search"
+    ;; only when ES store
+    (when (= "es" (p/get-in-global-properties [:ctia :store :target-record]))
+      (are [term check-fn expected desc] (let [response (helpers/get
+                                                         "ctia/target-record/search"
+                                                         :query-params {"query" term}
+                                                         :headers {"Authorization" "45c1f5e3f05d0"})]
+                                           (is (= 200 (:status response)))
+                                           (is (= expected (check-fn response)) desc))
+
+        "*"
+        (fn [r] (-> r :parsed-body first :targets first :type))
+        (-> target-record-sample :targets first :type)
+        "Searching for target-record:targets:type"
+
+        ;; TODO: Add more cases
+
+        ))))
+
+(deftest target-record-routes-test
+  (store/test-for-each-store
+   (fn []
+     (helpers/set-capabilities! "foouser" ["foogroup"] "user" auth/all-capabilities)
+     (whoami-helpers/set-whoami-response http/api-key "foouser" "foogroup" "user")
+     (entity-crud-test
+      {:entity           "target-record"
+       :example          new-target-record-maximal
+       :invalid-tests?   true
+       :update-tests?    true
+       :update-field     :source
+       :search-tests?    true
+       :additional-tests additional-tests
+       :headers          {:Authorization "45c1f5e3f05d0"}}))))
+
+(deftest target-record-pagination-test
+  (store/test-for-each-store
+   (fn []
+     (helpers/set-capabilities! "foouser" ["foogroup"] "user" auth/all-capabilities)
+     (whoami-helpers/set-whoami-response "45c1f5e3f05d0"
+                                         "foouser"
+                                         "foogroup"
+                                         "user")
+
+     (let [ids (helpers/post-entity-bulk
+                new-target-record-maximal
+                :target_records
+                30
+                {"Authorization" "45c1f5e3f05d0"})]
+
+       (field-selection/field-selection-tests
+        ["ctia/target-record/search?query=*"
+         (http/doc-id->rel-url (first ids))]
+        {"Authorization" "45c1f5e3f05d0"}
+        target-record/target-record-fields)
+
+       (pagination/pagination-test
+        "ctia/target-record/search?query=*"
+        {"Authorization" "45c1f5e3f05d0"}
+        target-record/target-record-fields)))))
+
+;; TODO: Fix
+#_(deftest target-record-metric-routes-test
```

Comment:
I co this branch and I manually tested the TargetRecord metric routes, it works.
Here there is a problem with the schema generation in the test helper.
2 solutions:
- figure out why the schema generation does not work and make it works, it is probably related to the schema generation that makes a field required for the generated entities (in orger to avoid aggregating empty values ;-)) https://github.com/threatgrid/ctia/blob/master/test/ctia/test_helpers/aggregate.clj#L180
```
(defn schema-enumerable-fields
  [schema fields]
  (->> (map (comp keyword first parse-field) fields)
       (st/select-keys schema)
       st/required-keys))
```
If I were you, I would first ensure that it produces the proper schema in your case (if it's actually the problem).
- Do not use the generative testing with a comment that explains why, and implement a non automatic testing (you can take some inspiration on the test helper anyway).

Reply:
I think I have fixed the test helpers, so they work with nested structures, see: https://github.com/threatgrid/ctia/pull/939/commits/8d5344b1748142a990a0c9d0e82d8dddfc2715c1
hope that doesn't break other things, I'm running tests now